web-api/src/business/useCases/trialSessions/getTrialSessionPlanningReportDataInteractor.ts

```diff
@@ -0,0 +1,276 @@
+import {
+  PreviousTerm,
+  TrialLocationData,
+} from '@shared/business/utilities/trialSessionPlanningReport/trialSessionPlanningReportDataTypes';
+import {
+  ROLE_PERMISSIONS,
+  isAuthorized,
+} from '../../../../../shared/src/authorization/authorizationClientService';
+import { RawTrialSession } from '@shared/business/entities/trialSessions/TrialSession';
+import {
+  SESSION_TYPES,
+  TRIAL_CITIES,
+  US_STATES,
+} from '../../../../../shared/src/business/entities/EntityConstants';
+import { ServerApplicationContext } from '@web-api/applicationContext';
+import { UnauthorizedError } from '@web-api/errors/errors';
+import { UnknownAuthUser } from '@shared/business/entities/authUser/AuthUser';
+import { capitalize, invert } from 'lodash';
+
+export const getTrialSessionPlanningReportDataInteractor = async (
+  applicationContext: ServerApplicationContext,
+  { term, year }: { term: string; year: number },
```

Comment:
Is it worth validating the term/year? Not a big deal, but sort of silly I can go to the trial session report for Atlantis -1234 and see a reasonable looking screen.

<img width="775" alt="Screenshot 2024-12-17 at 9 04 09 AM" src="https://github.com/user-attachments/assets/492cc1f7-472e-4613-9796-e224bf33e9f0" />